applications/convection_diffusion_application/custom_conditions/thermal_face.cpp

```diff
@@ -0,0 +1,430 @@
+// KRATOS ___ ___  _  ___   __   ___ ___ ___ ___
+//       / __/ _ \| \| \ \ / /__|   \_ _| __| __|
+//      | (_| (_) | .` |\ V /___| |) | || _|| _|
+//       \___\___/|_|\_| \_/    |___/___|_| |_|  APPLICATION
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Jordi Cotela
+//                   Riccardo Rossi
+//                   Ruben Zorrilla
+//
+
+
+#include "flux_condition.h"
+#include "includes/convection_diffusion_settings.h"
+#include "includes/variables.h"
+#include "utilities/integration_utilities.h"
+
+namespace Kratos
+{
+
+// Public Life Cycle //////////////////////////////////////////////////////////
+
+ThermalFace::ThermalFace(IndexType NewId, Geometry< Node<3> >::Pointer pGeometry):
+    Condition(NewId,pGeometry)
+{
+}
+
+ThermalFace::ThermalFace(
+    IndexType NewId,
+    Geometry< Node<3> >::Pointer pGeometry,
+    Properties::Pointer pProperties):
+    Condition(NewId,pGeometry,pProperties)
+{
+}
+
+ThermalFace::~ThermalFace()
+{
+}
+
+// Public Operations //////////////////////////////////////////////////////////
+
+Condition::Pointer ThermalFace::Create(
+    IndexType NewId,
+    NodesArrayType const& ThisNodes,
+    Properties::Pointer pProperties) const
+{
+    return Kratos::make_shared<ThermalFace>(NewId, GetGeometry().Create(ThisNodes), pProperties);
+}
+
+Condition::Pointer ThermalFace::Create(
+    IndexType NewId,
+    GeometryType::Pointer pGeom,
+    Properties::Pointer pProperties) const
+{
+    return Kratos::make_shared<ThermalFace>(NewId, pGeom, pProperties);
+}
+
+void ThermalFace::EquationIdVector(
+    EquationIdVectorType& rResult,
+    ProcessInfo& rCurrentProcessInfo)
+{
+    KRATOS_TRY
+
+    // Get unknown variable from convection diffusion settings
+    const ProcessInfo& r_process_info = rCurrentProcessInfo; // To ensure that Gets are threadsafe
+    ConvectionDiffusionSettings &r_conv_diff_settings = *(r_process_info[CONVECTION_DIFFUSION_SETTINGS]);
+    const Variable<double> &r_unknown_var = r_conv_diff_settings.GetUnknownVariable();
+
+    // Resize the equation ids. vector
+    auto &r_geometry = this->GetGeometry();
```

Comment:
const?
```suggestion
    const auto &r_geometry = this->GetGeometry();
```

Reply:
If `const`it complains when calling the `GetDof` method.